### PR TITLE
Implement project transfer API

### PR DESCRIPTION
**Pull request by drumm**
This implements another project transfer API: https://docs.gitlab.com/ce/api/projects.html#transfer-a-project-to-a-new-namespace

This was added by https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/20122, seems like permissions might be different from the other ways to transfer a project.

(For our instance of GitLab, this API returns normally. While `…->api('groups')->transfer(…)` does transfer the project, but the HTTP request times out.)

### CODE REVIEW

File: test/Gitlab/Tests/Api/TagsTest.php [hunk 1]
```diff
@@ -94,10 +94,10 @@ public function shouldCreateRelease($releaseName, $description, $expectedResult)
         $api = $this->getApiMock();
         $api->expects($this->once())
             ->method('post')
-            ->with('projects/1/repository/tags/' . urlencode($releaseName) . '/release', $params)
+            ->with('projects/1/repository/tags/' . str_replace(['/', '.'], ['%2F', '%2E'], $releaseName) . '/release', $params)
```
Comment (m1guelpf) [inline test/Gitlab/Tests/Api/TagsTest.php]: why this change?

Comment (drumm) [reply test/Gitlab/Tests/Api/TagsTest.php]: This was added by https://github.com/m4tthumphrey/php-gitlab-api/pull/368, rebasing so this pull request does not have extra changes.

`AbstractApi::encodePath()` explicitly encodes dots, this lines up with that for testing. https://github.com/m4tthumphrey/php-gitlab-api/blob/52965f4af56bb39dcb5393606b612424eab2a176/lib/Gitlab/Api/AbstractApi.php#L170
